### ISSUE TITLE
Bug: [no-unnecessary-qualifier] false positive for nested namespace

### ISSUE THREAD

**Issue by Jack-Works** [typescript-eslint/typescript-eslint#7192]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.1.6&fileType=.tsx&code=CYUwxgNghgTiAEA7KBbEBnADlMCCC8A3vAL4BQoksCyaWO%2BAdAEJFnzwgAemA9jABd4AgJ6YEALXgBeeAEYy5MqPHwpsvC0YSyQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0AjrCGMAZpUTR0UBdA6LwYAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
declare namespace A { }
declare namespace A.B {
  export type Z = 1
}

type Z = A.B.Z
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-qualifier": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error

### Actual Result

Reports error

**Comment by bradzacher** [comment]
Probably caused by our AST change in v6?
